Orange/statistics/util.py

```diff
@@ -355,22 +355,24 @@ def weighted_mean():
         return np.column_stack((
             np.nanmin(X, axis=0),
             np.nanmax(X, axis=0),
-            np.nanmean(X, axis=0) if not weighted else weighted_mean(),
-            np.nanvar(X, axis=0) if compute_variance else \
+            nanmean(X, axis=0) if not weighted else weighted_mean(),
+            nanvar(X, axis=0) if compute_variance else \
                 np.zeros(X.shape[1] if X.ndim == 2 else 1),
             nans,
             X.shape[0] - nans))
     elif is_sparse and X.size:
-        if compute_variance:
+        if compute_variance and weighted:
             raise NotImplementedError
 
         non_zero = np.bincount(X.nonzero()[1], minlength=X.shape[1])
         X = X.tocsc()
+        mean = nanmean(X, axis=0) if not weighted else weighted_mean()
         return np.column_stack((
             nanmin(X, axis=0),
             nanmax(X, axis=0),
-            nanmean(X, axis=0) if not weighted else weighted_mean(),
-            np.zeros(X.shape[1]),      # variance not supported
+            mean,
+            nanvar(X, axis=0) if compute_variance else \
+                np.zeros(X.shape[1] if X.ndim == 2 else 1),
             X.shape[0] - non_zero,
             non_zero))
```

Comment:
This seems very similar to the non sparse output. Could it be unified?

Reply:
The variance for sparse matrices does not support weights. If it did, the code would be the same. But hmm, if we put the condition out then, yeah, maybe we could unify. I'll try.

Reply:
Can't unify due to #6074. :) Otherwise I could.